test/DynamoPackagesUITests/DynamoPackagesUITests.cs

```diff
@@ -0,0 +1,172 @@
+            packageManagerCommands.Setup(t => t.Loader).Returns(this.Model.GetPackageManagerExtension().PackageLoader);
+
+            var package = new { asset_id = "12343", asset_name = "test" };
+            var version = new
+            {
+                contents = string.Empty,
+                contains_binaries = false,
+                node_libraries = new List<dynamic>(),
+                dependencies = string.Empty,
+                engine_version = "0.6.2.19362",
+                url = @"https://s3.amazonaws.com/greg-pkgs-dev/a5710c62-80bf-4d8a-92b5-28e242df9f4fgregPkg261.zip",
+                version = "2013.11.10",
+                file_id = "12133"
+            };
+            var dictPackage = package.GetType().GetProperties().ToDictionary(x => x.Name, x => x.GetValue(package, null));
+            var dictVersion = version.GetType().GetProperties().ToDictionary(x => x.Name, x => x.GetValue(version, null));
+            
+            
+            //1. Confirm Install Package Message Box
+            var packageManagerMessageBox = new Mock<IPackageManagerMessageBox>();
+            packageManagerMessageBox.Setup(t => t.ShowConfirmToInstallPackage(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<MessageBoxImage>())).Returns(MessageBoxResult.Cancel);
+
+            PackageManagerViewModel viewModel = new PackageManagerViewModel(packageManagerCommands.Object, packageManagerMessageBox.Object, "assets");
+            string returnValue = viewModel.PackageOnExecuted(dictPackage, dictVersion);
+            StringAssert.AreEqualIgnoringCase(returnValue, "cancel");
+
+            //2. Python Script Chaeck
+            packageManagerMessageBox.Setup(t => t.ShowConfirmToInstallPackage(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<MessageBoxImage>())).Returns(MessageBoxResult.OK);
+            packageManagerMessageBox.Setup(t => t.ShowPackageContainPythonScript(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<MessageBoxImage>())).Returns(MessageBoxResult.Cancel);
+            dictVersion["contains_binaries"] = true;
+            returnValue = viewModel.PackageOnExecuted(dictPackage, dictVersion);
+            StringAssert.AreEqualIgnoringCase(returnValue, "cancel");
+
+            //3. Check Dynamo Version
+            packageManagerMessageBox.Setup(t => t.ShowConfirmToInstallPackage(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<MessageBoxImage>())).Returns(MessageBoxResult.OK);
+            packageManagerMessageBox.Setup(t => t.ShowPackageContainPythonScript(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<MessageBoxImage>())).Returns(MessageBoxResult.OK);
+            dictVersion["contains_binaries"] = true;
+            returnValue = viewModel.PackageOnExecuted(dictPackage, dictVersion);
+            StringAssert.AreEqualIgnoringCase(returnValue, "12343,12133,test");
```

Comment:
I am not sure from where we get "12343,12133,test"? I think we should return a structured data from PackageOnExecuted(), so that it is clear to understand different pieces of data inside this string. It would also be good for handling different state returned from the method.